GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs

```diff
@@ -65,20 +69,126 @@ public CreateTestCaseRequestExtended BuildCaseRequest(Scenario scenario, IFeatur
                 WorkflowId = AddWorkflow(scenario, featureFile),
                 Description = AddDescription(scenario, featureFile),
                 Scenario = AddScenario(scenario, featureFile),
-                Tags = AddTags(scenario, featureFile)
+                Tags = AddTags(scenario, featureFile),
+                CustomFields = AddCustomFields(scenario, featureFile)
             },
             StepsAttachments = AddStepAttachments(scenario, featureFile)
         };
 
         return createTestCaseRequestExtended;
     }
 
+    private List<CustomFieldItem> AddCustomFields(Scenario scenario, IFeatureFile featureFile)
+    {
+        var result = new List<CustomFieldItem>();
+        AddFeatureCustomFiled(featureFile, result);
+        AddEpicCustomFiled(featureFile, result);
+        AddComponentCustomFiled(featureFile, scenario, result);
+        return result;
+    }
+
+    private void AddComponentCustomFiled(IFeatureFile featureFile, Scenario scenario, List<CustomFieldItem> result)
+    {
+        var settingsComponent = _allureTestOpsSettings.Component;
```

Comment:
let's move the variable declaration closer to the place where it is used

Reply:
Fixed